curl-helper.c

```diff
@@ -4862,13 +4839,44 @@ value caml_curl_multi_poll(value v_timeout_ms, value v_extra_fds, value v_multi)
   CAMLreturn(Val_bool(numfds != 0));
 }
 
+static int curlm_closesocket_cb(void *data, curl_socket_t socket)
+{
+  caml_leave_blocking_section();
+
+  CAMLparam0();
+  CAMLlocal1(camlResult);
+
+  ml_multi_handle* multi = (ml_multi_handle*)data;
+  int result = 0;
+  camlResult = caml_callback_exn(Field(multi->values, curlmopt_closesocket_function), Val_int(socket));
```

Comment:
Note that file descriptors are not represented as integers on Windows, so this code needs to be adapted in some way.

Reply:
You're right- I think I should use Val_socket here. Apologies, I don't often use windows. Changed.